zetaclient/evm/base_transaction_data.go

```diff
@@ -0,0 +1,179 @@
+package evm
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/rs/zerolog"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+	"github.com/zeta-chain/zetacore/zetaclient/interfaces"
+)
+
+const (
+	MinGasLimit = 100_000
+	MaxGasLimit = 1_000_000
+)
+
+// BaseTransactionData is a data structure containing input fields used to construct each type of transaction.
+// This is populated using cctx and other input parameters passed to TryProcessOutTx
+type BaseTransactionData struct {
```

Comment:
Instead of using `BaseTransactionData`, should we consider a more meaningful name, e.g. `OutboundKeysignData` or `OutboundTransactionData` to indicate the purpose of this data structure?